subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformIntegrationTest.groovy

```diff
@@ -1717,6 +1718,51 @@ Found the following transforms:
         output.count("Transforming") == 1
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/6156")
+    def "stops resolving dependencies of task when artifact transforms are encountered"() {
+        given:
+        buildFile << """
+            project(':lib') {
+                task jar(type: Jar) {
+                    destinationDir = buildDir
+                    archiveName = 'lib1.jar'
+                }
+
+                artifacts {
+                    compile jar
+                }
+            }
+
+            project(':app') {
+
+                dependencies {
+                    compile project(':lib')
+                }
+
+                ${configurationAndTransform()}
+
+                task dependent {
+                    dependsOn resolve
+                }
+            }
+
+            gradle.taskGraph.whenReady { taskGraph ->
+                taskGraph.allTasks.each { task ->
+                    task.taskDependencies.getDependencies(task).each { dependency ->
```

Comment:
Would it be interesting to print out `task`, too? So we would know that `dependent` depends on `resolve`?

Reply:
Let's do that...